Content.Shared/_Scp/Scp106/Systems/SharedScp106System.cs

```diff
@@ -41,53 +60,218 @@ public override void Initialize()
 
         SubscribeLocalEvent<Scp106Component, Scp106BackroomsActionEvent>(OnBackroomsDoAfter);
         SubscribeLocalEvent<Scp106Component, Scp106RandomTeleportActionEvent>(OnTeleportDoAfter);
+        SubscribeLocalEvent<Scp106PhantomComponent, Scp106BecomeTeleportPhantomActionEvent>(OnBecomeTeleportPhantomActionEvent);
 
         // Phantom
         SubscribeLocalEvent<Scp106PhantomComponent, Scp106ReverseAction>(OnScp106ReverseAction);
         SubscribeLocalEvent<Scp106PhantomComponent, Scp106LeavePhantomAction>(OnScp106LeavePhantomAction);
+        SubscribeLocalEvent<Scp106PhantomComponent, Scp106PassThroughAction>(OnScp106PassThroughAction);
 
+        SubscribeLocalEvent<Scp106PhantomComponent, Scp106ReverseActionEvent>(OnScp106ReverseActionEvent);
+        SubscribeLocalEvent<Scp106PhantomComponent, Scp106PassThroughActionEvent>(OnScp106PassThroughActionEvent);
+    }
+
+    private void OnScp106PassThroughAction(EntityUid uid,
+        Scp106PhantomComponent component,
+        Scp106PassThroughAction args)
+    {
+        if (args.Handled)
+            return;
+
+        if (!TryComp<FixturesComponent>(uid, out var fixturesComponent))
+            return;
+
+        foreach (var (id, fixture) in fixturesComponent.Fixtures)
+        {
+            _physics.SetCollisionMask(uid, id, fixture, (int) CollisionGroup.GhostImpassable);
+            _physics.SetCollisionLayer(uid, id, fixture, (int) CollisionGroup.GhostImpassable);
+        }
+
+        var doAfterEventArgs = new DoAfterArgs(EntityManager,
+            uid,
+            TimeSpan.FromSeconds(4),
+            new Scp106PassThroughActionEvent(),
+            uid)
+        {
+            BreakOnDropItem = false,
+            BreakOnMove = false,
+            BreakOnDamage = false,
+            BreakOnHandChange = false,
+            BreakOnWeightlessMove = false,
+        };
+
+        _doAfter.TryStartDoAfter(doAfterEventArgs);
+
+        args.Handled = true;
+    }
+
+    private void OnScp106PassThroughActionEvent(EntityUid uid,
+        Scp106PhantomComponent component,
+        Scp106PassThroughActionEvent args)
+    {
+        if (!TryComp<FixturesComponent>(uid, out var fixturesComponent))
+            return;
+
+        foreach (var (id, fixture) in fixturesComponent.Fixtures)
+        {
+            _physics.SetCollisionMask(uid, id, fixture, (int) CollisionGroup.SmallMobMask);
+            _physics.SetCollisionLayer(uid, id, fixture, (int) CollisionGroup.MobLayer);
+        }
+    }
+
+    private void OnBecomeTeleportPhantomAction(EntityUid uid,
+        Scp106Component component,
+        Scp106BecomeTeleportPhantomAction args)
+    {
+        if (component.IsContained)
+        {
+            _popup.PopupEntity("Ваши способности подавлены", uid, uid, PopupType.SmallCaution);
+            return;
+        }
+
+        if (component.Essence < 30)
+        {
+            _popup.PopupEntity($"Недостаточно {30 - component.Essence} эссенции!", uid, PopupType.Large);
```

Comment:
```suggestion
            _popup.PopupClient($"Недостаточно {30 - component.Essence} эссенции!", uid, PopupType.Large);
```

Попап клиент + нужна поддержка локализации